components/bl-weather-component/src/index.js

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from 'react';
+
+import ReactWeather, { useOpenWeather } from './lib/weather';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const UNITS = {
+  standard: { temperature: 'K', windSpeed: 'm/s' },
+  metric  : { temperature: '°C', windSpeed: 'm/s' },
+  imperial: { temperature: '°F', windSpeed: 'mph' },
+};
+const BASE_URL = 'https://api.openweathermap.org/data/2.5';
+
+export default function WeatherComponent({ component, elRef, settings }) {
+  const { classList, style, display, currentLocation, city, coords, forecast, unit, theme, lang } = component;
+  const { API_KEY } = settings;
+
+  const { location, weatherCity, hasGeoAccess } = useOptions(currentLocation, city, coords, API_KEY);
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div ref={ elRef } className={ cn('bl-customComponent-weather', ...classList) } style={ style }>
+      { hasGeoAccess && location?.lat && location.lng && (
+        <Weather
+          settings={ settings }
+          lat={ location.lat }
+          lon={ location.lng }
+          weatherCity={ weatherCity }
+          showForecast={ forecast }
+          unit={ unit }
+          theme={ theme }
+          lang={ lang }
+        />
+      ) }
+      { !hasGeoAccess && (
+        <p className="warn">To obtain weather data, you must allow access to your location.</p>
+      ) }
+    </div>
+  );
+}
+
+function Weather({ settings, lat, lon, weatherCity, showForecast, theme, unit, lang }) {
+  const key = settings.API_KEY;
+  const unitsLabels = UNITS[unit];
+  const { data, isLoading, errorMessage } = useOpenWeather({ key, lat, lon, lang, unit });
+
+  return (
+    <ReactWeather
+      isLoading={ isLoading }
+      errorMessage={ errorMessage }
+      data={ data }
+      lang={ lang }
+      locationLabel={ weatherCity }
+      unitsLabels={ unitsLabels }
+      showForecast={ showForecast }
+      theme={ theme }
+    />
+  );
+}
+
+function useOptions(currentLocation, city, coords, API_KEY) {
```

Comment:
Could we rename it to `useWeather`? I know we already have `useOpenWeather`, but as I see our hook is responsible for fetching data.

Also, I don't understand why we need to fetch data in this hook. Library has own methods to fetch data, and `useOpenWeather` is one of them. It's looks like you fetch data two times.

Reply:
Contact me please when returns to this part again.